segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+from pathlib import Path
+import random
+from typing import Union, Tuple
+import os.path
+import math
+import copy
+
+import h5py
+import cv2
+import numpy as np
+import imageio_ffmpeg as mpg
+
+
+def downsample_h5_video(video_path: Union[str, Path], input_fps: int,
+                     output_fps: int, strategy: str = 'random') -> np.ndarray:
+    """
+    Function to down-sample an array stored in h5 format to a desired FPS.
+    Follows the strategy specified in the function call, current strategies are
+    (random, maximum, average). Random takes a random sample of one frame from
+    window, maximum takes the maximum frame in window, average takes the mean
+    of window.
+    Args:
+        video_path: The path to the input video in str or Path format
+        input_fps: The FPS or Hz of the input video
+        output_fps: The desired output FPS or Hz
+        strategy: The down-sampling startegy to follow
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if os.path.exists(video_path):
+        path_parts = str(video_path).split('.')
+        extension = path_parts[len(path_parts) - 1]
+        if extension == 'h5':
+            with h5py.File(video_path, 'r') as open_video:
+                video = open_video['data'][:]
+                return _down_sample_array(full_array=video,
+                                          input_fps=input_fps,
+                                          output_fps=output_fps,
+                                          strategy=strategy)
+        else:
+            raise AttributeError('Video path provided not an h5 file, please'
+                                 'provide an h5 file')
+    else:
+        raise FileNotFoundError('Path specified doesnt exist or is not accessible')
+
+
+def _down_sample_array(full_array: np.ndarray, input_fps: int,
+                       output_fps: int, strategy: str = 'random') -> np.ndarray:
+    """
+    Function to down-sample a numpy array to a desired FPS. Follows the
+    strategy specified in the function call, current strategies are (random,
+    maximum, average). Random takes a random sample of one frame from window,
+    maximum takes the maximum frame in window, average takes the mean of window.
+    Args:
+        full_array: The video array in numpy format
+        input_fps: The FPS or Hz of the input video
+        output_fps: The desired output FPS or Hz
+        strategy: The down-sampling startegy to follow
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if output_fps > input_fps:
+        raise ValueError('Output FPS cannot be greater than input FPS')
+    sampling_ratio = int(input_fps / output_fps)
```

Comment:
I think this will accumulate a time-dilation. For example 31Hz -> 4Hz, this will always be 8. The time axis of the downsampled movie won't be the time axis of the original movie. A way around this is to use `np.array_split` or `np.split` to make arrays of unequal size.

Reply:
Fixed time dilation I think, I'll need some confirmation this is fixed when I update it.